test/parallel/test-http2-settings-unsolicited-ack.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const http2 = require('http2');
+const net = require('net');
+const http2util = require('../common/http2');
+const Countdown = require('../common/countdown');
+
+// Test that an unsolicited settings ack is ignored.
+
+const kSettings = new http2util.SettingsFrame();
+const kSettingsAck = new http2util.SettingsFrame(true);
+
+const server = http2.createServer();
+let client;
+
+const countdown = new Countdown(3, () => {
+  client.destroy();
+  server.close();
+});
+
+server.on('stream', common.mustNotCall());
+server.on('session', common.mustCall((session) => {
+  session.on('remoteSettings', common.mustCall(() => countdown.dec()));
+}));
+
+server.listen(0, common.mustCall(() => {
+  client = net.connect(server.address().port);
+  client.on('connect', common.mustCall(() => {
+    client.write(http2util.kClientMagic, () => {
+      client.write(kSettings.data, () => {
+        // Give time to receive the servers settings frame
+        // so that the first ack below is actually expected.
+        setImmediate(() => {
```

Comment:
I fear this might be time sensitive. Can we avoid it anyhow? Maybe add a comment here

Reply:
We can with a bit of dancing between the client and server. Will take another pass.

Reply:
actually, no dancing required. pretty straight forward :-)